bundle/edu.gemini.pot/src/main/java/edu/gemini/spModel/gemini/gmos/GmosOiwfsGuideProbe.java

```diff
@@ -130,8 +129,9 @@ public PatrolField getPatrolField() {
     @Override
     public double calculateVignetting(final ObsContext ctx,
                                       final edu.gemini.spModel.core.Coordinates guideStarCoordinates) {
-        final ProbeArmGeometry probeArmGeometry       = GmosOiwfsProbeArm.instance();
-        return ScalaGuideProbeUtil.calculateVignetting(ctx, guideStarCoordinates, probeArmGeometry);
+        if (Site.GN.equals(ctx.getSite().getOrNull()))
+            ScalaGuideProbeUtil.calculateVignetting(ctx, guideStarCoordinates, GmosNorthOiwfsProbeArm$.MODULE$);
+        return ScalaGuideProbeUtil.calculateVignetting(ctx, guideStarCoordinates, GmosSouthOiwfsProbeArm$.MODULE$);
```

Comment:
Ooops! Haha. How utterly stupid of me, and good point.
Hmmm... I'm not sure that your suggestion would work based on all the type parameters.